### PR TITLE
moving multilabel functions

### PR DESCRIPTION
**Pull request by aditya1503**
Moving multilabel functions (one hot encoders), and also fixing Pytest warnings

### CODE REVIEW

File: tests/test_latent_algebra.py [hunk 1]
```diff
@@ -27,29 +27,34 @@ def test_latent_py_ps_inv():
     ps, py, inv = latent_algebra.compute_ps_py_inv_noise_matrix(s, nm)
     assert all(abs(np.dot(inv, ps) - py) < 1e-3)
     assert all(abs(np.dot(nm, py) - ps) < 1e-3)
```
Comment (aditya1503) [inline tests/test_latent_algebra.py:29]: A future version of Pytest will not support return statements, 
hence separating this into separate functions

File: cleanlab/count.py [hunk 2]
```diff
@@ -1381,6 +1381,6 @@ def _get_confident_thresholds_multilabel(
     for class_num, (label, pred_prob) in enumerate(zip(y_one.T, pred_probs.T)):
         pred_probabilitites = stack_complement(pred_prob)
         confident_thresholds[class_num] = get_confident_thresholds(
-            pred_probs=pred_probabilitites, labels=labels
+            pred_probs=pred_probabilitites, labels=label
```
Comment (aditya1503) [inline cleanlab/count.py:1384]: can't believe I missed this, thanks to warnings

Comment (elisno) [reply cleanlab/count.py:1384]: Nice catch!

**Review by elisno — APPROVED** [review]
LGTM! :shipit: 

I have just one thought

### "one-hot" encoding
This might be a misnomer if we intend to only use `int2onehot` & `onehot2int` exclusively for multi-label classification while it's also useful for multi-class classification. It would be closer to "multi-hot" encoding (bad name ☹️).

We can look at this at another time.

**Comment by jwmueller** [comment]
Agreed that int2onehot & onehot2int are confusingly named.

Let's at the very least update their docstrings to clearly state  these are binzarized formats of the labels in multi-hot vector.

**Comment by aditya1503** [comment]
> Agreed that int2onehot & onehot2int are confusingly named.
> 
> Let's at the very least update their docstrings to clearly state these are binzarized formats of the labels in multi-hot vector.

https://github.com/cleanlab/cleanlab/pull/515/commits/e27356b3e7939e88775921eb34d3eac6b9a7e164

File: cleanlab/internal/multilabel_utils.py [hunk 3]
```diff
@@ -64,3 +64,37 @@ def get_onehot_num_classes(labels, pred_probs=None):
             "wrong format for labels, should be a list of list[indices], please check the documentation in find_label_issues for further information"
         )
     return y_one, num_classes
+
+
+def int2onehot(labels) -> np.ndarray:
+    """Convert list of lists to a onehot matrix for multi-labels which is
+     a binzarized format of the labels in a multi-hot vector.
```
Comment (jwmueller) [inline cleanlab/internal/multilabel_utils.py:71]: Typo

File: cleanlab/internal/multilabel_utils.py [hunk 4]
```diff
@@ -64,3 +64,37 @@ def get_onehot_num_classes(labels, pred_probs=None):
...
+      All integers from 0,1,...,K-1 must be represented."""
+
+    from sklearn.preprocessing import MultiLabelBinarizer
+
+    mlb = MultiLabelBinarizer()
+    return mlb.fit_transform(labels)
+
+
+def onehot2int(onehot_matrix) -> list:
+    """Convert a onehot matrix for multi-labels to a list of lists of ints
```
Comment (jwmueller) [inline cleanlab/internal/multilabel_utils.py:87]: Clarification to Docstring should be added here too

File: cleanlab/internal/multilabel_utils.py [hunk 5]
```diff
@@ -64,3 +64,38 @@ def get_onehot_num_classes(labels, pred_probs=None):
...
+
+    from sklearn.preprocessing import MultiLabelBinarizer
+
+    mlb = MultiLabelBinarizer()
+    return mlb.fit_transform(labels)
+
+
+def onehot2int(onehot_matrix) -> list:
+    """Convert a onehot matrix for multi-labels which is
+     a binarized format to a list of lists of ints.
```
Comment (jwmueller) [inline cleanlab/internal/multilabel_utils.py:88]:
```suggestion
    """Convert a multi-hot matrix for multi-label classification (which is specifically a onehot-like matrix of 0s and 1s in a binarized format with multiple 1s allowed per row) to a list of lists of ints that can be used with other cleanlab functions for multi-label classification.
```

File: cleanlab/count.py [hunk 6]
```diff
@@ -1415,6 +1415,6 @@ def _get_confident_thresholds_multilabel(
     for class_num, (label, pred_prob) in enumerate(zip(y_one.T, pred_probs.T)):
         pred_probabilitites = stack_complement(pred_prob)
         confident_thresholds[class_num] = get_confident_thresholds(
```
Comment (jwmueller) [inline cleanlab/count.py:1417]:
```suggestion
    for class_num, (label_for_class, pred_prob_for_class) in enumerate(zip(y_one.T, pred_probs.T)):
        pred_probs_binary = stack_complement(pred_prob_for_class)
        confident_thresholds[class_num] = get_confident_thresholds(
```

File: cleanlab/count.py [hunk 7]
```diff
@@ -1415,6 +1415,6 @@ def _get_confident_thresholds_multilabel(
...
-            pred_probs=pred_probabilitites, labels=labels
+            pred_probs=pred_probabilitites, labels=label
         )
```
Comment (jwmueller) [inline cleanlab/count.py:1419]:
```suggestion
            pred_probs=pred_probs_binary, labels=label_for_class
        )
```

Comment (jwmueller) [reply cleanlab/count.py:1419]: The bug appeared because of not using precise enough variable names. We want to be as precise as possible in the variable names for code clarity and avoiding bugs like this (and should never have typos in variable names like `pred_probabilitites`)

File: cleanlab/internal/multilabel_utils.py [hunk 8]
```diff
@@ -64,3 +64,34 @@ def get_onehot_num_classes(labels, pred_probs=None):
             "wrong format for labels, should be a list of list[indices], please check the documentation in find_label_issues for further information"
         )
     return y_one, num_classes
+
+
+def int2onehot(labels, K) -> np.ndarray:
+    """Convert list of lists to a onehot matrix for multi-labels
+    Parameters
```
Comment (jwmueller) [inline cleanlab/internal/multilabel_utils.py:71]:
```suggestion
    """Convert multi-label classification `labels` from a ``List[List[int]]`` format to a onehot matrix.
    This returns a binarized format of the labels as a multi-hot vector for each example, where the entries in this vector are 1 for each class that applies to this example and 0 otherwise.
    
    Parameters
```

Comment (jwmueller) [reply cleanlab/internal/multilabel_utils.py:71]: note there should always be more spacing between the initial docstring description and `Parameters` or `Returns` or `Note` or any other special header.

File: cleanlab/internal/multilabel_utils.py [hunk 9]
```diff
@@ -64,3 +64,34 @@ def get_onehot_num_classes(labels, pred_probs=None):
...
+
+    from sklearn.preprocessing import MultiLabelBinarizer
+
+    mlb = MultiLabelBinarizer(classes=range(K))
+    return mlb.fit_transform(labels)
+
+
+def onehot2int(onehot_matrix) -> list:
+    """Convert a onehot matrix for multi-labels to a list of lists of ints
+    Parameters
```
Comment (jwmueller) [inline cleanlab/internal/multilabel_utils.py:87]:
```suggestion
    """Convert multi-label classification `labels` from a onehot matrix format to a ``List[List[int]]`` format that can be used with other cleanlab functions.
    
    Parameters
```

File: cleanlab/internal/multilabel_utils.py [hunk 10]
```diff
@@ -64,3 +64,34 @@ def get_onehot_num_classes(labels, pred_probs=None):
...
+    ----------
+    onehot_matrix: 2D np.ndarray of 0s and 1s
+      A one hot encoded matrix representation of multi-labels.
+    Returns
```
Comment (jwmueller) [inline cleanlab/internal/multilabel_utils.py:91]:
```suggestion
    onehot_matrix: 2D np.ndarray of 0s and 1s
      A matrix representation of multi-label classification labels in a binarized format as a multi-hot vector for each example. 
      The entries in this vector are 1 for each class that applies to this example and 0 otherwise.
    
    Returns
```

File: cleanlab/internal/multilabel_utils.py [hunk 11]
```diff
@@ -64,3 +64,34 @@ def get_onehot_num_classes(labels, pred_probs=None):
             "wrong format for labels, should be a list of list[indices], please check the documentation in find_label_issues for further information"
         )
     return y_one, num_classes
+
+
+def int2onehot(labels, K) -> np.ndarray:
```
Comment (jwmueller) [inline cleanlab/internal/multilabel_utils.py:69]: args should have type annotations

File: cleanlab/internal/multilabel_utils.py [hunk 12]
```diff
@@ -64,3 +64,34 @@ def get_onehot_num_classes(labels, pred_probs=None):
...
+    K: int
+      The number of classes."""
+
+    from sklearn.preprocessing import MultiLabelBinarizer
+
+    mlb = MultiLabelBinarizer(classes=range(K))
+    return mlb.fit_transform(labels)
+
+
+def onehot2int(onehot_matrix) -> list:
```
Comment (jwmueller) [inline cleanlab/internal/multilabel_utils.py:85]: args should have type annotations

Comment (jwmueller) [reply cleanlab/internal/multilabel_utils.py:85]: should we make the return type more explicit as `List[List[int]]` ?

File: cleanlab/internal/multilabel_utils.py (hunk 11 above)
Comment (aditya1503) [reply cleanlab/internal/multilabel_utils.py]: Added: https://github.com/aditya1503/cleanlab/blob/627b297f286e0d4b0289b3c9f724ad39ae2866a1/cleanlab/internal/multilabel_utils.py#L72

File: cleanlab/internal/multilabel_utils.py (hunk 12 above)
Comment (aditya1503) [reply cleanlab/internal/multilabel_utils.py]: Added: https://github.com/aditya1503/cleanlab/blob/42eaf5dc794eb1b1c28fb45263f6aec73e8c2cd2/cleanlab/internal/multilabel_utils.py#L90
